functorch/functorch/dim/__init__.py

```diff
@@ -0,0 +1,170 @@
+import torch
+from typing import Union, Sequence
+import inspect
+import dis
+from .tree_map import tree_flatten, tree_map
+from .wrap_type import wrap_type
+from functorch._C import dim as _C
+_C._patch_tensor_class()
```

Comment:
To check my understanding: the monkey patching for torchdim happens on `import functorch.dim` or `from functorch.dim import ...` and does not happen if someone just calls `import functorch`, right?

Reply:
Correct, I was careful that no patching happens unless the dim package is imported. A follow up is to either write a more in-depth `__torch_function__` for getitem/setitem, or to simply merge dim handling into getitem/setitem code.

Reply:
(y) thanks for confirming